client/wfprev-war/src/main/angular/src/app/components/map/map.component.ts

```diff
@@ -429,4 +440,23 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     });
   }
 
+  // Close any open popup and fully remove the active marker so no DOM is orphaned
+  private teardownActiveUI(): void {

```

Comment:
you are performing cleanup and then nulling a reference here. Can we add try/ fiannly here to ensure this.activeMarker is still nulled If any of those cleanup calls throw? 
ie:
if (this.activeMarker) {
  try {
    this.activeMarker.unbindPopup();

    if (this.markersClusterGroup?.hasLayer(this.activeMarker)) {
      this.markersClusterGroup.removeLayer(this.activeMarker);
    } else {
      this.activeMarker.remove();
    }
  } finally {
    this.activeMarker = null;
  }
}

Reply:
sure, done